### DISCUSSION TITLE
Moving to more responsive/reactive design

### DISCUSSION THREAD

**Discussion by ChefDaddyJay** [jdwilkin4/contrast-ratio-repo#153] [General]
This was just something I thought needed its own thread. The app looks good on both my desktop and mobile, but there are some things that could probably make user interaction a little smoother. The biggest one I've noticed is that the page doesn't react until the user changes focus away from the input field. Making this happen on submit or (even better) on change, would make things a little easier, especially when the user is experimenting with multiple color themes.

Just my thoughts. Any other ideas that might help things flow better for the user?

**Comment by Dorfieeee** [comment]
We can add `setTimeout` with some reasonable timer on the inputs `onChange` event listener.
When user types the first letter, timer will start counting 500ms for example and if he does not start typing again within that time frame, validation will kick in, if the user types within the time frame, the timer resets and starts counting again.

**Reply by sparkO7** [reply]
I like this!

**Comment by sparkO7** [comment]
This is true and we were definitely gonna work on that.
Since you brought this up now, we can add it as an issue, so someone can start working on it.
